lib/crypto.js

```diff
@@ -562,9 +557,10 @@ exports.pbkdf2 = function(password,
 
 
 exports.pbkdf2Sync = function(password, salt, iterations, keylen, digest) {
-  if (typeof digest === 'undefined') {
-    digest = undefined;
-    pbkdf2DeprecationWarning();
+  if (digest === undefined) {
+    // Formerly Deprecation code DEP0009
+    throw new TypeError(
+      'The "digest" argument is required and must not be undefined');
```

Comment:
Ditto.  Out of curiosity, is there a reason you can't move the check to the pbkdf2 function below?  That would avoid the duplication.